### PR TITLE
Update proj extension to include grid

### PR DESCRIPTION
**Pull request by alexgleith**
**Related Issue(s):** #756

**Proposed Changes:**

1. Incorporate the proposed grid extension changes into the projections extension
2. Include two new properties, `proj:shape` and `proj:transform`

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

File: extensions/projection/README.md [hunk 1]
```diff
@@ -52,20 +58,41 @@ If the data does not have a CRS, such as in the case of non-rectified imagery wi
...
+**proj:centroid** - Coordinates representing the centroid of the item in the asset data CRS.  Coordinates are
 defined in latitude and longitude, even if the data coordinate system does not use lat/long.
 
+**proj:shape** - An array of integers that represents the number of pixels in the most common pixel grid used by the item's assets.
+The number of pixels should be specified in Y, X order. If the shape is defined in an item's properties it is used as
+the default shape for all assets that don't have an overriding shape.
+
+**proj: transform** - Linear mapping from pixel coordinate space (Pixel, Line) to projection coordinate space (Xp, Yp). It is a `3x3` matrix stored as a flat array of 9 elements in row major order. Since the last row is always `0,0,1` it can be omitted, in which case only 6 elements are recorded. This mapping can be obtained from GDAL([`GetGeoTransform`](https://gdal.org/api/gdaldataset_cpp.html#_CPPv4N11GDALDataset15GetGeoTransformEPd)) or the Rasterio ([`Transform`](https://rasterio.readthedocs.io/en/stable/api/rasterio.io.html#rasterio.io.BufferedDatasetWriter.transform)).
```
Comment (m-mohr) [inline extensions/projection/README.md:79]: There's a space in the field name ;-)

File: extensions/projection/README.md [hunk 2]
```diff
@@ -52,20 +58,41 @@ If the data does not have a CRS, such as in the case of non-rectified imagery wi
...
+  [1 ]   [0 ,  0,  1]   [1    ]
+```
+
+If the transform is defined in an item's properties it is used as the default transform for all assets that don't have an overriding transform.
+
+### Item [`Asset Object`](../../item-spec/item-spec.md#asset-object) fields
+
+| Field Name | Type     | Description                                  |
+| ---------- | -------- | -------------------------------------------- |
+| proj:shape    | \[number] | See item description.  |
```
Comment (m-mohr) [inline extensions/projection/README.md:93]: Above it's said that the shape consists of integers, so I'd propose to use \[integer].

File: extensions/projection/json-schema/schema.json [hunk 3]
```diff
@@ -85,6 +85,24 @@
...
                 }
               }
+            },
+            "proj:shape":{
+              "title":"Shape",
+              "type":"array",
+              "minItems":2,
+              "maxItems":2,
+              "items":{
+                "type":"number"
```
Comment (m-mohr) [inline extensions/projection/json-schema/schema.json:95]: Above it's said that the shape consists of integers, so I'd propose to use type: integer.

File: extensions/projection/README.md [hunk 4]
```diff
@@ -9,30 +9,36 @@
...
 | ---------------- | ------------------------ | ----------- |
-| proj:epsg        | integer\|null   | **Required** [EPSG code](http://www.epsg-registry.org/) of the datasource |
+| proj:epsg        | integer \|null  | **Required** [EPSG code](http://www.epsg-registry.org/) of the datasource |
 | proj:wkt2        | string \|null   | [WKT2](http://docs.opengeospatial.org/is/12-063r5/12-063r5.html) string representing the Coordinate Reference System (CRS) that the `proj:geometry` and `proj:bbox` fields represent |
 | proj:projjson    | [PROJJSON Object](https://proj.org/usage/projjson.html) \|null   | PROJJSON object representing the Coordinate Reference System (CRS) that the `proj:geometry` and `proj:bbox` fields represent |
 | proj:geometry    | [Polygon Object](https://geojson.org/schema/Polygon.json)  | Defines the footprint of this Item. |
 | proj:bbox        | \[number]       | Bounding box of the Item in the asset CRS in 2 or 3 dimensions. |
 | proj:centroid    | Centroid Object | Coordinates representing the centroid of the Item in the asset CRS |
+| proj:shape       | \[number]       | Number of pixels in Y and X directions for the default grid |
```
Comment (m-mohr) [inline extensions/projection/README.md:40]: Below it's said that the shape consists of integers, so I'd propose to use [integer].

File: extensions/projection/README.md (hunk 1 above)
Comment (alexgleith) [reply extensions/projection/README.md:79]: Oops!

File: extensions/projection/json-schema/schema.json (hunk 3 above)
Comment (alexgleith) [reply extensions/projection/json-schema/schema.json:95]: Good idea

File: extensions/projection/README.md (hunk 4 above)
Comment (alexgleith) [reply extensions/projection/README.md:40]: 👍

**Comment by alexgleith** [comment]
Hey @m-mohr, thanks for the review. I think I've resolved all your comments. If you can check the schema, as it's a bit confusing as to where to put the asset level example. I think I have it in the right place!

**Comment by matthewhanson** [comment]
@m-mohr @alexgleith I don't think there should be a schema for the asset....because this extension only adds Item properties.   It's a not-yet-created asset specific metadata extension that would allow these Item properties to be specified at the asset level.   I'm not sure how schema validation would happen there (perhaps allow any Item properties to be in the asset), but regardless it doesn't need to be handled in this extension.

**Comment by m-mohr** [comment]
@matthewhanson I disagree, the extension defines 

> Item [`Asset Object`](../../item-spec/item-spec.md#asset-object) fields

So there should be validation for it OR the section mentioned above should be removed.

**Comment by matthewhanson** [comment]
Ah you are right, thought that had been changed.

@alexgleith the fields should be moved out of the Asset object and in only the Item properties.

I'll get a PR up by end of this week with the Asset specific metadata extension

**Comment by alexgleith** [comment]
Ah ok. Do we think that we should be explicit in the documentation that the item-level information can be over-ridden?

It's hard to discover if it isn't explicit in this extension.

**Comment by alexgleith** [comment]
Ok, I've pushed a change to the schema and readme, but left a note in on the two properties that says "If the shape is defined in an item's properties it is used as the default shape for all assets that don't have an overriding shape."

Does that make sense?

**Comment by matthewhanson** [comment]
Yes, I think this is one of the places it should say it can be overridden at the asset level because these are some of the fields for which that makes sense to do.

**Comment by alexgleith** [comment]
> Looks mostly good, but some issues with number/integer. See the comments for details. Additionally, there's no JSON Schema for the Assets.

I think this has been resolved now, @m-mohr

**Pull request merged**